syntax = "proto3";

message IntMat {
    int32 num_dim = 1;
    repeated int32 shape = 2;
    repeated int32 data = 3;
}

message FloatMat {
    int32 num_dim = 1;
    repeated int32 shape = 2;
    repeated float data = 3;
}

message ByteBuffer {
    repeated bytes buffer = 1;
}

// in .glb format
message Model3D {
    int32 size = 1;
    bytes data = 3;
}

message GarmentReconstructionRequest {
    FloatMat garment_img = 1;
    FloatMat pose = 2;
}

service GarmentReconstruction {
    rpc reconstruct (GarmentReconstructionRequest) returns (stream Model3D) {}
}

service PoseDetection {
    rpc getPose (ByteBuffer) returns (FloatMat) {}
}